parser.js

```diff
@@ -0,0 +1,119 @@
+var CompileError = require("./compileerror");
+var solc = require("solc");
+
+// Clean up after solc.
+var listeners = process.listeners("uncaughtException");
+var solc_listener = listeners[listeners.length - 1];
+
+if (solc_listener) {
+  process.removeListener("uncaughtException", solc_listener);
+}
+
+module.exports = {
+  parse: function(body, fileName) {
+    // Here, we want a valid AST even if imports don't exist. The way to
+    // get around that is to tell the compiler, as they happen, that we 
+    // have source for them (an empty file).
+
+    var fileName = fileName || "ParsedContract.sol";
+
+    var solcStandardInput = {
+      language: "Solidity",
+      sources: {
+        [fileName]: {
+          content: body
+        }
+      },
+      settings: {
+        outputSelection: {
+          "Contract.sol": {
```

Comment:
Should this match `ParsedContract.sol` above?

Reply:
Whoops, good catch.